transmit/direct_transmit.go

```diff
@@ -147,6 +160,12 @@ func (d *DirectTransmission) Stop() error {
 	}
 	d.transmitQueues = nil
 	d.shutdownWG.Wait()
+
+	if d.batchPool != nil {
+		d.batchPool.Wait()
+	}
+	d.batchPool = nil
```

Comment:
I'm curious why we want to assign `nil` value to the `batchPool` on shutdown?

Reply:
Because that way if something wasn't shut down properly, it will probably panic. Which is better than it silently dropping events and never being noticed forever.